File: Java/Curso-Em-Video/POO/Aula06/ControleRemoto.java
package Aula06;

public class ControleRemoto implements Controlador
{
    private int volume;
    private boolean ligado;
    private boolean tocando;

    public ControleRemoto() 
    {
        this.setVolume(50);
        this.setLigado(true);;
        this.setTocando(false);;
    }

    @Override
    public void ligar() 
    {
        this.setLigado(true);
    }

    @Override
    public void desligar() 
    {
        this.setLigado(false);
    }

    @Override
    public void abrirMenu() 
    {
        System.out.println("--------- MENU ---------");
        System.out.println("Está ligado? "+ this.isLigado());
        System.out.println("Está tocando? "+ this.isTocando());
        System.out.print("Volume: "+ this.getVolume());

        for(int i = 0; i <= this.getVolume(); i += 10)
        {
            System.out.print("|");
        }

        System.out.println("");
    }

    @Override
    public void fecharMenu() 
    {
        System.out.println("Fechar menu...");
    }

    @Override
    public void maisVolume() 
    {
        if(this.isLigado())
        {
            this.setVolume(this.getVolume() + 5);
        }

        else
        {
            System.out.println("Impossível aumentar o volume");
        }
    }

    @Override
    public void menosVolume() 
    {
        if(!this.isLigado())
        {
            this.setVolume(this.getVolume() - 5);
        }

        else
        {
            System.out.println("Impossível diminuir o volume");
        }
    }

    @Override
    public void ligarMudo() 
    {
        if(this.isLigado() && this.getVolume() > 0)
        {
            this.setVolume(0);
        }
    }

    @Override
    public void desligarMudo() 
    {
        if(this.isLigado() && this.getVolume() > 0)
        {
            this.setVolume(50);
        }
    }

    @Override
    public void play() 
    {
        if(this.isLigado() && !(this.isTocando()))
        {
            this.setTocando(true);
        }

        else
        {
            System.out.println("Não consegui reproduzir");
        }
    }

    @Override
    public void pause() 
    {
        if(this.isLigado() && this.isTocando())
        {
            this.setTocando(false);
        }

        else
        {
            System.out.println("Não consegui pausar");
        }
    }

    public int getVolume() 
    {
        return volume;
    }

    public void setVolume(int volume) 
    {
        this.volume = volume;
    }

    public boolean isLigado() 
    {
        return ligado;
    }

    public void setLigado(boolean ligado) 
    {
        this.ligado = ligado;
    }

    public boolean isTocando() 
    {
        return tocando;
    }

    public void setTocando(boolean tocando) 
    {
        this.tocando = tocando;
    }
}

File: PHP/Curso-Em-Video/POO/Aula06/ControleRemoto.php
<?php
require_once 'Controlador.php';
class ControleRemoto implements Controlador
{
    private $volume;
    private $ligado;
    private $tocando;

    public function __construct()
    {
        $this -> volume = 50;
        $this -> ligado = false;
        $this -> tocando = false;
    }

    private function getVolume()
    {
        return $this->volume;
    }

    private function setVolume($volume)
    {
        $this->volume = $volume;

        return $this;
    }

    private function getLigado()
    {
        return $this->ligado;
    }

    private function setLigado($ligado)
    {
        $this->ligado = $ligado;

        return $this;
    }

    private function getTocando()
    {
        return $this->tocando;
    }

    private function setTocando($tocando)
    {
        $this->tocando = $tocando;

        return $this;
    }

    public function ligar()
    {
        $this -> setLigado(true);
    }

    public function desligar()
    {
        $this -> setLigado(false);
    }

    public  function abrirMenu()
    {
        echo "<p>----- MENU -----</p>";
        echo "<br>Esta ligado?: ". ($this -> getLigado()?"Sim":"Não");
        echo "<br>Esta tocando?: ". ($this -> getTocando()? "Sim":"Não");
        echo "<br>Volume: ". ($this -> getVolume());
        for($i = 0; $i <= $this -> getVolume(); $i += 10)
        {
            echo "|";
        }

        echo "<br>";
    }

    public function fecharMenu()
    {
        "<br>Fechando Menu...";
    }

    public function maisVolume()
    {
        if($this -> getLigado())
        {
            $this -> setVolume($this -> getVolume() + 10);
        }

        else
        {
            echo "<p>ERRO! Não posso aumentar o volume</p>";
        }
    }

    public function menosVolume()
    {
        if($this -> getLigado())
        {
            $this -> setLigado($this -> getVolume() - 10);
        }

        else
        {
            echo "<p>ERRO! Não posso diminuir o volume</p>";
        }
    }

    public function ligarMudo()
    {
        if($this -> getLigado() && $this -> getVolume() > 0)
        {
            $this -> setVolume(0);
        }
    }

    public function desligarMudo()
    {
        if($this -> getLigado() && $this -> getVolume() == 0)
        {
            $this -> setVolume(50);
        }
    }

    public function play()
    {
        if($this -> getLigado() && !($this -> getTocando()))
        {
            $this ->setTocando(true);
        }
    }

    public function pause()
    {
        if($this -> getLigado() && $this -> getTocando())
        {
            $this -> setTocando(false);
        } 
    }
}